### PR TITLE
Added php-task library

### PR DESCRIPTION
**Pull request by wachterjohannes**
Added `php-task` library.

Basic PR: https://github.com/sulu/sulu/pull/3037

Depends on https://github.com/php-task/TaskBundle/pull/28 (is merged)

#### TODO 
- [x] tests
- [x] code-style

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -38,7 +38,9 @@
         "doctrine/data-fixtures": "1.1.*",
         "layershifter/tld-extract": "^1.1",
         "mtdowling/cron-expression": "^1.1",
-        "ramsey/uuid": "^3.1"
+        "ramsey/uuid": "^3.1",
+        "php-task/task-bundle": "dev-master",
```
Comment (chirimoya) [inline composer.json:42]: ??? ;-)

File: src/Sulu/Bundle/AutomationBundle/EventListener/TaskEventListener.php [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+use Task\Scheduler\TaskSchedulerInterface;
+use Task\Storage\TaskExecutionRepositoryInterface;
+use Task\Storage\TaskRepositoryInterface;
+use Task\TaskInterface as PHPTaskInterface;
+use Task\TaskStatus;
+
+/**
+ * Handles php-task library integration.
+ */
+class TaskEventListener
```
Comment (chirimoya) [inline src/Sulu/Bundle/AutomationBundle/EventListener/TaskEventListener.php]: explicit call of php-task

File: src/Sulu/Bundle/AutomationBundle/Exception/TaskHandlerNotSupportedException.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+     */
+    private $task;
+
+    /**
+     * @param TaskHandlerInterface $taskHandler
+     * @param TaskInterface $task
+     */
+    public function __construct(TaskHandlerInterface $taskHandler, TaskInterface $task)
+    {
+        parent::__construct(sprintf("Task-Handler '%s' is not supported.", get_class($taskHandler)));
```
Comment (chirimoya) [inline src/Sulu/Bundle/AutomationBundle/Exception/TaskHandlerNotSupportedException.php]: singel quotes

**Review by chirimoya — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (wachterjohannes) [reply composer.json:42]: i will create a new version before we merge this feature into develop

**Comment by wachterjohannes** [comment]
@chirimoya fixed (:

File: src/Sulu/Bundle/AutomationBundle/Tasks/Scheduler/TaskSchedulerInterface.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+     * @throws TaskExpiredException
+     */
+    public function reschedule(TaskInterface $task);
+
+    /**
+     * Unschedules given task.
+     *
+     * @param TaskInterface $task
+     */
+    public function unschedule(TaskInterface $task);
```
Comment (chirimoya) [inline src/Sulu/Bundle/AutomationBundle/Tasks/Scheduler/TaskSchedulerInterface.php]: `remove`

File: src/Sulu/Bundle/AutomationBundle/Tasks/Scheduler/TaskSchedulerInterface.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    public function schedule(TaskInterface $task);
+
+    /**
+     * Reschedule given task.
+     *
+     * @param TaskInterface $task
+     *
+     * @throws TaskExpiredException
+     */
+    public function reschedule(TaskInterface $task);
```
Comment (chirimoya) [inline src/Sulu/Bundle/AutomationBundle/Tasks/Scheduler/TaskSchedulerInterface.php:36]: `update`

**Review by chirimoya — CHANGES_REQUESTED** [review]

**Comment by chirimoya** [comment]
ACK

**Pull request merged**